test/pytest/test_xpedite/test_profiler/test_profiler.py

```diff
@@ -149,31 +101,28 @@ def test_probe_states(capsys):
   """
   import cPickle as pickle
   from xpedite.types.probe import compareProbes
-  profileInfo = loadProfileInfo(DATA_DIR, os.path.join(DATA_DIR, 'profileInfo.py'), REMOTE)
-  probes = []
-  probeMap = {}
-  baselineProbeMap = {}
-  with capsys.disabled():
-    probes = loadProbes(FIX_DECODER_BINARY, profileInfo, TXN_COUNT, THREAD_COUNT, REMOTE, WORKSPACE)
-    for probe in probes:
-      probeMap[probe.sysName] = probe
-
-  with open(os.path.join(DATA_DIR, 'probeCmdBaseline.pkl'), 'r') as probeFileHandle:
-    baselineProbes = pickle.load(probeFileHandle)
-    for probe in baselineProbes:
-      baselineProbeMap[probe.sysName] = probe
-
-  assert len(probes) == len(baselineProbes)
-  findDiff(probeMap, baselineProbeMap)
-  for probe in probeMap.keys():
-    assert compareProbes(probeMap[probe], baselineProbeMap[probe])
+  for scenarios in SCENARIO_LOADER:
+    with scenarios as scenarios:
+      probes = []
```

Comment:
Do we need this variable ? Can't we can directly build the map and do validations on the map ?